sql/sql_insert.cc

```diff
@@ -1640,6 +1640,11 @@ int vers_insert_history_row(TABLE *table)
   // Set Sys_end to now()
   table->vers_update_end();
 
+  Field *row_start= table->vers_start_field();
+  Field *row_end= table->vers_end_field();
+  if (row_start->cmp(row_start->ptr, row_end->ptr) != -1)
```

Comment:
1. how can be **row_start > row_end**?
2. It is better to add all comparison operators to `Field` class: **==, !=, >, <, >=, <=**.

Reply:
> how can be row_start > row_end?

```sql
create or replace table t1 (
pk int,
a char(3),
b char(3),
primary key(pk))
engine=myisam with system versioning character set ucs2;
insert into t1 (pk) values (0);
connect  con1,localhost,root,,test;
update t1 set a = 'bar' where pk=0;
connection default;
update t1 set b = 'foo' where pk=0;
connection con1;
disconnect con1;
connection default;
select row_start<row_end, row_start, row_end, a, b from t1 for system_time all;
row_start<row_end	row_start	row_end	a	b
1	2018-02-01 12:10:29.336350	2038-01-19 06:14:07.999999	bar	foo
1	2018-02-01 12:10:29.335442	2018-02-01 12:10:29.336354	NULL	NULL
0	2018-02-01 12:10:29.336354	2018-02-01 12:10:29.336350	NULL	foo
```

For me it looks like a timer issues. `row_start`, `row_end` has a pretty big error for short-lived rows. That's a case where InnoDB `trx_id` is much better.


> It is better to add all comparison operators to Field class: ==, !=, >, <, >=, <=.

Well, maybe. But I don't think it's a scope of this patch. Such stuff should be in a separate PR with many code changes which are not related to system-versioning.

Reply:
1. It is Windows-crutch code.
2. Don't make things complex: just add operators and use them.

Reply:
> It is Windows-crutch code.

What do you mean?